math_test.go

```diff
@@ -276,6 +282,18 @@ func runModAddSubNegTest(t *testing.T, c *Curve) {
 	assert.True(t, bagain.Equals(b))
 }
 
+func runMulTest(t *testing.T, c *Curve) {
+	rng, err := c.Rand()
+	assert.NoError(t, err)
+
+	r := c.NewRandomZr(rng)
```

Comment:
can we please add a D/H style test as well - like that a^b^c = a^c^b = a^bc?

Reply:
Let me know if it's OK